### PR TITLE
refactor setdata and add add different test cases.

### PR DESCRIPTION
**Pull request by stevenhua0320**

### CODE REVIEW

**Review by sbillinge — COMMENTED** [review]
So the same thing here. Make a matrix of all the things you want to test. Write it as text in comments (delete everything there, or better yet, start a new PR), we can discuss it, then we can go from there.  For now the function should only be a docstring and something that allows the test to fail and not error.

**Comment by stevenhua0320** [comment]
> So the same thing here. Make a matrix of all the things you want to test. Write it as text in comments (delete everything there, or better yet, start a new PR), we can discuss it, then we can go from there. For now the function should only be a docstring and something that allows the test to fail and not error.

Hi Simon, I add the text comments in test files for what I want to check for behavior in `setdata` function. Probably after confirming what we want to test I would open a separate PR for the setdata test.

File: src/diffpy/srmise/tests/test_dataclusters.py [hunk 1]
```diff
@@ -13,6 +13,18 @@ def test_clear():
     assert actual == expected
 
 
+# In the set data test, we test for these cases.
+# (1) x and y are non-empty array values, and res is positive (the most generic case)
```
Comment (sbillinge) [inline src/diffpy/srmise/tests/test_dataclusters.py:17]: Good. This test should cover what is desired for the other attributes that are determined from these inputs.

**Review by sbillinge — COMMENTED** [review]
Good, please see comments

**Pull request closed without merging**